### PR TITLE
WIP: Splash

### PR DESCRIPTION
**Pull request by raigex**
Splash screen
- Add db handling and api calls as necessary

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,6 +30,17 @@ buildscript {
                 arch_paging            : "android.arch.paging:runtime:${paging_version}",
                 arch_common            : "android.arch.paging:common:${arch_version}",
                 arch_rx                : "android.arch.lifecycle:reactivestreams:${arch_version}",
+                retrofit               : 'com.squareup.retrofit2:retrofit:2.3.0',
+                retrofit_rx            : 'com.jakewharton.retrofit:retrofit2-rxjava2-adapter:1.0.0',
+                retrofit_scalars       : 'com.squareup.retrofit2:converter-scalars:2.1.0',
+                retrofit_moshi         : 'com.squareup.retrofit2:converter-moshi:2.1.0',
+                sendbird               : 'com.sendbird.sdk:sendbird-android-sdk:3.0.54',
```
Comment (agrosner) [inline build.gradle]: @raigex don't add unnecessary dependencies.

File: db/src/main/kotlin/edu/artic/db/ApiModule.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+@Module
+abstract class ApiModule {
+
+    @Module
+    companion object {
+        @JvmStatic
+        @Provides
+        @Singleton
+        @Named("BlobRetrofit")
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/ApiModule.kt]: Please rename this. BlobRetrofit is weird. and make a constant for the Named injection.

File: db/src/main/kotlin/edu/artic/db/models/AricMapAnnotation.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package edu.artic.db.models
+
+data class AricMapAnnotation(
+val title : String,
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/AricMapAnnotation.kt]: indentation is a little messy here. use the autoformat on android studio.

File: db/src/main/kotlin/edu/artic/db/models/AricMapAnnotation.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package edu.artic.db.models
+
+data class AricMapAnnotation(
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/AricMapAnnotation.kt]: Spelling errors :)

File: db/src/main/kotlin/edu/artic/db/models/ArticDataObject.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package edu.artic.db.models
+
+data class ArticDataObject(
+        val image_server_url: String,
+        val data_api_url: String,
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/ArticDataObject.kt]: Snake case vs camel case. I would use the moshi `@Json` annotation to clean up the names to camelcase

File: db/src/main/kotlin/edu/artic/db/models/ArticObject.kt [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package edu.artic.db.models
+
+class ArticObject(
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/ArticObject.kt]: `data` class here too?

File: db/src/main/kotlin/edu/artic/db/models/ArticTour.kt [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+val floor : String,
+val image_filename : String,
+val image_url : String,
+val image_filemime : String,
+val image_filesize : String,
+val image_width : String,
+val image_height : String,
+//val thumbnail_crop_rect : [],
+val thumbnail_full_path : String,
+//val large_image_crop_rect : [],
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/ArticTour.kt]: commented out code. either leave these out, or add comment please.

File: db/src/main/kotlin/edu/artic/db/models/ArticTour.kt [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+val selector_number : String ,
+val description : String,
+val description_html : String,
+val tour_dates : TourDate,
+val intro : String,
+val intro_html : String,
+val tour_duration : String,
+val tour_audio : String,
+val category : Any, // TODO: figure out what this is
+val stops : List<Any>, //TODO: figure out what this is
```
Comment (agrosner) [inline db/src/main/kotlin/edu/artic/db/models/ArticTour.kt]: Look at iOS for inspiration instead. :)

File: splash/src/androidTest/java/edu/artic/splash/ExampleInstrumentedTest.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
```
Comment (agrosner) [inline splash/src/androidTest/java/edu/artic/splash/ExampleInstrumentedTest.java]: remove these generated files whenever adding new module.

File: splash/src/test/java/edu/artic/splash/ExampleUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class ExampleUnitTest {
```
Comment (agrosner) [inline splash/src/test/java/edu/artic/splash/ExampleUnitTest.java]: same here. remove.

File: viewmodel/src/main/kotlin/edu/artic/viewmodel/BaseViewModelFragment.kt [hunk 11]
```diff
@@ -3,5 +3,9 @@ package edu.artic.viewmodel
 import edu.artic.ui.BaseFragment
 
 abstract class BaseViewModelFragment<TViewModel : BaseViewModel> : BaseFragment() {
```
Comment (agrosner) [inline viewmodel/src/main/kotlin/edu/artic/viewmodel/BaseViewModelFragment.kt]: your'e missing the injection stuff here too. need to port it over from activity so you can inject viewmodels in fragments tooo.

File: db/src/main/kotlin/edu/artic/db/models/ArticDataObject.kt (hunk 5 above)
Comment (raigex) [reply db/src/main/kotlin/edu/artic/db/models/ArticDataObject.kt]: will do so before merge

File: db/src/main/kotlin/edu/artic/db/models/ArticTour.kt (hunk 8 above)
Comment (raigex) [reply db/src/main/kotlin/edu/artic/db/models/ArticTour.kt]: Will do so before merging ( hence WIP status)

**Pull request merged**